### DISCUSSION TITLE
How to get semantic tokens intellisense in theme files

### DISCUSSION THREAD

**Discussion by veloware** [chakra-ui/chakra-ui#8286] [Q&A] [answered]
I have generated my typescript types for semantic tokens using the CLI, and works as expected in most cases, but im not sure how to get intellisense when building out a theme file for a new custom component, e.g - 

```
const { defineMultiStyleConfig, definePartsStyle } = createMultiStyleConfigHelpers([
  'container',
  'header',
  'body',
  'footer',
]);

const baseStyle = definePartsStyle({
  container: {
    bgColor: '$bg.', // <!-- not getting any TS help to complete the token name
  },
});
```

the only way I can can get TS help for semantic token suggestions is to use `as` to cast the type of the object

```
const baseStyle = definePartsStyle({
  container: {
    bgColor: '$bg.', // <!-- now I get suggestions for tokens beginning with `$bg.`
  } as BoxProps,
});
```

but I was hoping to use the chakra factory for this `container` element instead of a Box component, e.g. 

```
<chakra.div __css={styles.container}>
```
so in this case casting to `BoxProps` for an element that isn't a `Box` seems a bit hacky.

Any suggestions?

**Comment by 0x33dm** [comment]
following this.

**Comment by segunadebayo** [comment, marked as answer]
`defineMultiStyleConfig` / `definePartsStyle` is v2. Those helpers do not pick up generated token types.

In v3, define a recipe or slot recipe and run `npx @chakra-ui/cli typegen`. Token names complete in the recipe object.
